### ISSUE TITLE
macOS server

### ISSUE THREAD

**Issue by prnxdev** [multitheftauto/mtasa-blue#792]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
Create executable server for macOS

**Describe alternatives you've considered**
None

**Additional context**
I'm MacOS user. I'm also a full stack JS programmer. I really prefer unix-like system because of simplicity and integrated terminal so I often work on MacOS. If I want to make some cool stuff on MTA I have to switch to Windows PC. Not a big thing but sometimes it means a lot - switching cables, changing keyboard habits etc. I know that there is steam version of GTA:SA. I suggest that make server and client executables for MacOS system or though only server - I can still emulate Windows on VM.

**Comment by qaisjp** [comment]
This is something I've been dabbling in and out of for a while. It's a lot of work! 

I'll create a separate issue for the client (later) as that involves a lot of additional development, and is something I've worked towards in the past.

**Comment by prnxdev** [comment]
Is there an issue for client? Actually what is to be done for the client to work?

**Comment by qaisjp** [comment]
I have written instructions to get Wine working on macOS: https://gist.github.com/qaisjp/778ed99ac8a8dcb93f797a28ba5f8303. The same instructions (with a few tweaks) works on Linux.

A future "client" would package either Wine, Wineskin or Proton into a nice installer.

Historical work on the latter is on [wiki macOS], but I don't think it is up to date, so we should probably not use Wineskin.

[wiki macOS]:  https://wiki.multitheftauto.com/wiki/MacOS

**Comment by StiviiK** [comment]
Any updates on the macOS server build? As the compile seems to work fine and only the net-Module is missing. Or am I missing something else which delays macOS server builds?

Btw. @qaisjp is the Idea with a macOS client still a thing?

**Comment by qaisjp** [comment]
Server: yes we're blocked on net module. 

Client: there are working instructions here https://gist.github.com/qaisjp/778ed99ac8a8dcb93f797a28ba5f8303 but I can't get it to work on Catalina because it only runs 64bit software (and wine64 is unreliable on macOS, in fact impossible until recently)
